### DISCUSSION TITLE
Invalid hashicorp constraint for version = "0.1.0-beta.0"

### DISCUSSION THREAD

**Discussion by gberche-orange** [renovatebot/renovate#21332] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm observing the following unexpected warning about version constraint, whereas a fixed version `0.1.0-beta.0` is instead specified. Is this because this `0.1.0-beta.0` version is not complying with semvers and hence misleadingly interpreted as a range ?

```
WARN: Invalid hashicorp constraint
{
  "baseBranch": "feature-bumps",
  "constraint": "0.1.0-beta.0",
  "element": "0.1.0-beta.0"
}
DEBUG: Dependency siderolabs/talos has unsupported/unversioned value 0.1.0-beta.0 (versioning=hashicorp)
{
  "baseBranch": "feature-bumps"
}
```

For a valid terraform required-providers.tf file:

```json
terraform {

  required_version = ">= 0.12"
  #experiments      = [module_variable_optional_attrs]
  
  required_providers {
    
    vsphere = {
      source = "hashicorp/vsphere"
      version = "2.1.1"
    }
    
    nsxt = {
      source = "vmware/nsxt"
      version = "3.2.6"
    }
    
    credhub = {
      source = "orange-cloudfoundry/credhub"
      version = "0.14.2"
    }

    talos = {
      source = "siderolabs/talos"
      version = "0.1.0-beta.0"
    }

    
  }
}
```

See associated provider version https://github.com/siderolabs/terraform-provider-talos/releases/tag/v0.1.0-beta.0

### What you tried so far.

A workaround is to manually bump to 0.1.0

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Will be fixed by https://github.com/renovatebot/renovate/pull/21334

**Reply by gberche-orange** [reply]
thanks @rarkins !
